pkg/storaged/stratis/pool.jsx

```diff
@@ -64,6 +65,283 @@ const _ = cockpit.gettext;
+            result.push({ slot: null, keydesc: key_desc[1][1] });
+    } else {
+        let key_descs = pool.KeyDescriptions;
+        /* The KeyDescriptions property itself has type "v". For a V1
+           pool this variant carries a value of type "b(bs)" just as
+           with the r6 API (see above). For a V2 pool, it carries a
+           "a(us)" value, an array of "slot number plus key
+           description".
+
+           Since the property itself has type "v", we need unwrap this
+           variant explicitly.
+
+           HACK - https://github.com/stratis-storage/stratisd/issues/3895
+           However, change notifications drop the variant
+           wrapping for some reason, so we only do the unwrapping when
+           "key_descs" is indeed a variant.
+         */
+        if ("t" in key_descs)
+            key_descs = key_descs.v;
+        if (pool.MetadataVersion == 1) {
+            if (key_descs[0] && key_descs[1][0])
+                result.push({ slot: null, keydesc: key_descs[1][1] });
+        } else if (pool.MetadataVersion == 2) {
+            for (const key_desc of key_descs)
+                result.push({ slot: key_desc[0], keydesc: key_desc[1] });
+        }
+    }
+
+    return result;
+}
+
+/* Clevis information is returned as an array with elements of type
+ *
+ *   { slot: number | null, pin: "tang", url: string }
+ * | { slot: number | null, pin: string }
+ *
+ * A V1 pool will have at most one entry in the array, and it will
+ * have `slot == null`.  V2 pools have zero or more entries with
+ * numbered slots.
+ */
+
+function get_clevis_infos(pool) {
+    const result = [];
+
+    if (!pool.Encrypted)
+        return result;
+
+    if (client.stratis_interface_revision < 8) {
+        const clevis_info = pool.ClevisInfo;
+        /* The ClevisInfo property has type "(b(b(ss)))".
+
+           The first "b" is false when the encryption of the pool is
+           inconsistent (not all block devices have the same LUKS
+           header). We pretend that the pool doesn't have any Clevis
+           infos in that case.
+
+           The second "b" is true when the pool has a Clevis info, and
+           the two "s" are the pin and its JSON encoded configuration,
+           respectively.
+
+           Thus, we detect an object of shape
+
+               [ true, [ true, [ "...", "..." ] ] ]
+
+           and fish the strings out of it.
+
+           Cockpit only knows about the "tang" pin in detail, so we
+           detect that as well and only then decode the configuration
+           in order to extract the url from it.
+         */
+        if (clevis_info[0] && clevis_info[1][0]) {
+            if (clevis_info[1][1][0] == "tang") {
+                const config = JSON.parse(clevis_info[1][1][1]);
+                result.push({ slot: null, pin: "tang", url: config.url });
+            } else {
+                result.push({ slot: null, pin: clevis_info[1][1][0] });
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-21927-76c8d326-20250825-170718-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)